src/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java

```diff
@@ -1066,4 +1076,24 @@ public String getJcrOutputPath() {
     }
   }
 
+  protected int checkAndGetUserInputQueryLimit( final Object userInputQueryLimit, final int reportQueryLimit ) {
```

Comment:
Just one thing: 
If - on the URL - I pass in a user-limit that is higher than the limit that is already hard-coded on the report, this code will not give precedence to the hard-coded limit. So if I have a top-10 report, I can happily extend it to a Top-1000 report, and thus see data to which I am not supposed to have access. This is a great way to download a list of all customers, or do other evil things.